jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java

```diff
@@ -927,6 +928,34 @@ default int size()
         return size;
     }
 
+    /**
+     * @param fields the {@link HttpFields} to convert to a {@link Map}.
+     * @return an unmodifiable {@link Map} representing the contents of the {@link HttpFields}.
+     */
+    static Map<String, List<String>> asMap(HttpFields fields)
+    {
+        Map<String, List<String>> headers = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+        for (HttpField f : fields)
+        {
+            if (!headers.containsKey(f.getName()))
+            {
+                HttpHeader header = f.getHeader();
+                List<String> values = header == null ? fields.getValuesList(f.getName()) : fields.getValuesList(header);
+                headers.put(f.getName(), Collections.unmodifiableList(values));
+            }
+        }
+        return Collections.unmodifiableMap(headers);
```

Comment:
The other thing to consider here is just wrapping as a HttpFieldsMap and then returning the asImmutable version of it.   That may be simpler and easier than doing the full copy here?

Reply:
I considered doing this, but I would have to create a separate implementation of the `HttpFieldsMap` which takes `HttpFields` instead of `HttpFields.Mutable`.

Reply:
I added this.

Reply:
Nice!